app.R

```diff
@@ -0,0 +1,237 @@
+    setBackgroundColor("LightSkyBlue"),
+    
+    # create Heading
+    titlePanel(h1("Get your personal weather forecast", align = "center")),
+    br(), 
+    br(), 
+    
+    
+    sidebarLayout(
+        
+        sidebarPanel(
+            
+            style = "background-color: EggWhite;",
+        
+            # insert search field
+            fluidRow(column(width = 12, 
+                            align = "center",
+                            textInput(inputId = "location", h3("Please enter a location"),
+                                      value = "Amsterdam"))
+                     ),
+            
+            # insert select box 
+            # this is not useful yet but I intend to use it later, if time allows.
+            fluidRow(column(width = 12,
+                            align = "center",
+                            selectInput("cur_hour_day", "Please indicate the type of forecast you are looking for",
+                                        c("current weather", "hourly forecast", "daily forecast")))
+                     ),
+            
+            # insert search button 
+            # has to be used with "eventReactive()"
+            fluidRow(column(width = 12, actionButton(inputId = "search", "Search"),
+                            align = "center")
+                     ),
+            br(),
+            br(),
+            
+            # display weather GIF: 
+            fluidRow(
+                column(width = 12, 
+                       align = "center", 
+                       HTML("<div style='height: 150px;'>"),
+                       imageOutput("weather_gif"), 
+                       HTML("</div>"))
+            )
+        ),
+        
+        mainPanel(
+            
+            # display chosen location and weather icon: 
+            fluidRow(
+                column(width = 6,
+                       offset = 3,
+                       verbatimTextOutput("my_output_location"),
+                       align = "center"),
+                column(width = 3,
+                       align = "center", 
+                       HTML("<div style='height:40px;'>"),
+                       imageOutput("weather_icon"), 
+                       HTML("</div>"))
+            ), 
+            br(), 
+            
+            # display weather image: 
+            fluidRow(
+                column(width = 12, 
+                       align = "center", 
+                       HTML("<div style='height: 100px;'>"),
+                       withSpinner(imageOutput("weather_image"), type = 1), 
+                       HTML("</div>"))
+            ),
+            br(), 
+            br(),
+            
+            # display current general weather forecast
+            fluidRow(
+                column(width = 6, 
+                       offset = 3,
+                       verbatimTextOutput("current_weather"),
+                       align = "center")
+            ),
+            

```

Comment:
You could make your code less redundant by adding these four fluidRow()s into one apply function. (Something like apply(c(“current_temp”, “sunrise”), function(i) { fluidRow() with all the output settings here}))